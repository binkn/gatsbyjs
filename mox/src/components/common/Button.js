import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'

// Themes
import { Colors } from '../../themes/Colors'
import { px2Rem } from '../../themes/Mixins'

const Button = (props) => (
  <button
    style={props.customStyle}
    onClick={props.onPress}
    className={props.className}
  >
    {props.text}
  </button>
)

export default styled(Button)`
  height: ${px2Rem(30)};
  border-radius: ${px2Rem(7)};
  border: ${px2Rem(1)} solid transparent;
  font-size: ${px2Rem(14)};
  padding: ${px2Rem(6)} ${px2Rem(12)};
  margin: ${px2Rem(15)};
  text-transform: uppercase;
  cursor: pointer;

  :hover {
    opacity: 0.4;
  }

  ${props => props.type === 'default' && css`
    color: ${Colors.white};
    background-color: ${Colors.aquamarine};
  `};

  ${props => props.type === 'primary' && css`
    background-color: ${Colors.turquoise};
    color: ${Colors.white};
    border-color: ${Colors.white};
  `};

  ${props => props.type === 'secondary' && css`
    background-color: ${Colors.white};
    color: ${Colors.black};
  `};

  ${props => props.type === 'third' && css`
    background-color: ${Colors.black};
    color: ${Colors.white};
  `};
`
Button.propTypes = {
  text: PropTypes.string,
  onPress: PropTypes.func,
  customStyle: PropTypes.object,
  className: PropTypes.string
}

Button.defaultProps = {
  text: '',
  onPress: () => {},
  customStyle: {},
  className: ''
}
