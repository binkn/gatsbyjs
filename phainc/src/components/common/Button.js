import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'

// Themes
import { Colors } from '../../themes/Colors'
import { SIZE, TYPE } from '../../themes/Typography'

// components
import Image from '../common/Image'

const Button = React.memo(props => {
  const { className, customStyle, text, onPress, src } = props
  const { buttonStyle, imgStyle } = customStyle

  return (
    <button
      type='button'
      style={buttonStyle}
      className={className}
      onClick={onPress}
    >
      {text ? text : ''}
      {
        src &&
        <Image
          src={props.src}
          alt={props.alt}
          customStyle={imgStyle}
        />
      }
    </button>
  )
})

export default styled(Button)`
  font-family: ${props => props.fontFamily || TYPE.roboto};
  font-weight: bold;
  font-size: ${props => props.fontSize || SIZE.xSmall};
  border: 2px solid ${Colors.concrete};
  background-color: transparent;
  border-radius: 3px;
  cursor: pointer;
  outline: none;
  padding: 12px 35px;
  letter-spacing: 0.3px;

  &:hover {
    color: ${Colors.carnation};
  }

  ${props => props.type === 'light' && css`
    color: ${Colors.white};

    &:hover {
      color: ${Colors.mineShaft};
      background-color: ${Colors.white};
    }
  `};

  ${props => props.type === 'dark' && css`
    color: ${Colors.mineShaft};

    &:hover {
      color: ${Colors.white};
      background-color: ${Colors.carnation};
    }
  `};

  ${props => props.type === 'glow' && css`
    background-image: linear-gradient(to right, ${Colors.violetRed}, ${Colors.salmon});
    color: ${Colors.white};

    &:hover {
      background-image: none;
      background-color: ${Colors.salmon};
    }
  `};

  ${props => props.type === 'transform' && css`
    color: ${Colors.carnation};
    border-color: ${Colors.carnation};
    position: relative;

    &:hover {
      color: ${Colors.white};
      background-image: linear-gradient(to right, ${Colors.violetRed}, ${Colors.salmon});
      box-shadow: 0 0px 15px ${Colors.primary};
      transform: translateY(-5px);
      border: none;
    }
  `};

  ${props => props.type === 'none-border' && css`
    background: none;
    color: ${Colors.malibu};
    border: none;
  `};
`

Button.propTypes = {
  text: PropTypes.string,
  customStyle: PropTypes.object,
  className: PropTypes.string,
  fontFamily: PropTypes.string,
  fontSize: PropTypes.string,
  src: PropTypes.string,
  onPress: PropTypes.func
}

Button.defaultProps = {
  text: '',
  customStyle: {},
  className: '',
  fontFamily: '',
  fontSize: '',
  src: '',
  onPress: () => {}
}
