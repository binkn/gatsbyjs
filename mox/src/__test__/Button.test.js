import React from 'react'
import { configure, shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import 'jest-styled-components'

// Setup adapter to work with enzyme 3.2.0
configure({ adapter: new Adapter() })

// Components
import Button from '../components/common/Button'

describe('Button test', () => {

  const props = {
    text: 'About us',
    onPress: jest.fn()
  }

  let component = shallow(<Button {...props}/>)

  it('Snapshot Button component', () => {
    const renderedValue = renderer.create(
      <Button {...props} />
    ).toJSON()
    expect(renderedValue).toMatchSnapshot()
  })

  it('Snapshot Button component with type default', () => {
    const renderedValue = renderer.create(
      <Button {...props} type='default' />
    ).toJSON()
    expect(renderedValue).toMatchSnapshot()
  })

  it('Snapshot Button component type primary', () => {
    const renderedValue = renderer.create(
      <Button {...props} type='primary' />
    ).toJSON()
    expect(renderedValue).toMatchSnapshot()
  })

  it('Snapshot Button component type secondary', () => {
    const renderedValue = renderer.create(
      <Button {...props} type='secondary' />
    ).toJSON()
    expect(renderedValue).toMatchSnapshot()
  })

  it('Snapshot Button component type third', () => {
    const renderedValue = renderer.create(
      <Button {...props} type='third' />
    ).toJSON()
    expect(renderedValue).toMatchSnapshot()
  })

  it('should be defined', () => {
    expect(Button).toBeDefined()
  })

  it('render Button component', () => {
    expect(component.length).toEqual(1)
  })
})
