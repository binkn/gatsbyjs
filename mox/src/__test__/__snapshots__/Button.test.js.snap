// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Snapshot Button component 1`] = `
.c0 {
  height: 1.875rem;
  border-radius: 0.4375rem;
  border: 0.0625rem solid transparent;
  font-size: 0.875rem;
  padding: 0.375rem 0.75rem;
  margin: 0.9375rem;
  text-transform: uppercase;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.4;
}

<button
  className="c0"
  onClick={[MockFunction]}
  style={Object {}}
>
  About us
</button>
`;

exports[`Button test Snapshot Button component type primary 1`] = `
.c0 {
  height: 1.875rem;
  border-radius: 0.4375rem;
  border: 0.0625rem solid transparent;
  font-size: 0.875rem;
  padding: 0.375rem 0.75rem;
  margin: 0.9375rem;
  text-transform: uppercase;
  cursor: pointer;
  background-color: #3BD0D5;
  color: #fff;
  border-color: #fff;
}

.c0:hover {
  opacity: 0.4;
}

<button
  className="c0"
  onClick={[MockFunction]}
  style={Object {}}
>
  About us
</button>
`;

exports[`Button test Snapshot Button component type secondary 1`] = `
.c0 {
  height: 1.875rem;
  border-radius: 0.4375rem;
  border: 0.0625rem solid transparent;
  font-size: 0.875rem;
  padding: 0.375rem 0.75rem;
  margin: 0.9375rem;
  text-transform: uppercase;
  cursor: pointer;
  background-color: #fff;
  color: #000;
}

.c0:hover {
  opacity: 0.4;
}

<button
  className="c0"
  onClick={[MockFunction]}
  style={Object {}}
>
  About us
</button>
`;

exports[`Button test Snapshot Button component type third 1`] = `
.c0 {
  height: 1.875rem;
  border-radius: 0.4375rem;
  border: 0.0625rem solid transparent;
  font-size: 0.875rem;
  padding: 0.375rem 0.75rem;
  margin: 0.9375rem;
  text-transform: uppercase;
  cursor: pointer;
  background-color: #000;
  color: #fff;
}

.c0:hover {
  opacity: 0.4;
}

<button
  className="c0"
  onClick={[MockFunction]}
  style={Object {}}
>
  About us
</button>
`;

exports[`Button test Snapshot Button component with type default 1`] = `
.c0 {
  height: 1.875rem;
  border-radius: 0.4375rem;
  border: 0.0625rem solid transparent;
  font-size: 0.875rem;
  padding: 0.375rem 0.75rem;
  margin: 0.9375rem;
  text-transform: uppercase;
  cursor: pointer;
  color: #fff;
  background-color: #5BF2FF;
}

.c0:hover {
  opacity: 0.4;
}

<button
  className="c0"
  onClick={[MockFunction]}
  style={Object {}}
>
  About us
</button>
`;
